homework/tests/homework.spec.js

```diff
@@ -0,0 +1,230 @@
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('1.5');
+    });
+
+    test('Adding 4 and k results in an error message', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('4');
+      await page.locator('#number2Field').type('k');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+  
+      var error = await page.textContent('#errorMsgField');
+      await expect(error).toContain('Number 2 is not a number');
+    });
+
+    test('Concatenating 4 and k results in 4k', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('4');
+      await page.locator('#number2Field').type('k');
+      await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('4k');
+    });
+
+    test('Subtract 4 from k results in an error message', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('4');
+      await page.locator('#number2Field').type('k');
+      await page.selectOption('#selectOperationDropdown', {label: 'Subtract'});
+      await page.locator('#calculateButton').click();
+  
+      var error = await page.textContent('#errorMsgField');
+      await expect(error).toContain('Number 2 is not a number');
+    });
+
+    test('Multiplying 4 by k results in an error message', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('4');
+      await page.locator('#number2Field').type('k');
+      await page.selectOption('#selectOperationDropdown', {label: 'Multiply'});
+      await page.locator('#calculateButton').click();
+  
+      let error = await page.textContent('#errorMsgField');
+      await expect(error).toContain('Number 2 is not a number');
+    });
+
+    test('Dividing 4 from k results in an error message', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('4');
+      await page.locator('#number2Field').type('k');
+      await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+      await page.locator('#calculateButton').click();
+  
+      let error = await page.textContent('#errorMsgField');
```

Comment:
Declaring a variable which is used only once - is good when the function is quite long and if we'd combine it it would be hard to read. So this can be used for readability purposes. [Or if we want to combine several constants in one place, so they could potentially be reused in later tests development]

Getting errorTextContent is not long ar complex function, so readability won't suffer. 
Don't use variables which are used only once later in the test. Please adjust.